rio_tiler/models.py

```diff
@@ -67,3 +71,77 @@ class Metadata(Info):
     """Dataset metadata and statistics."""
 
     statistics: Dict[str, ImageStatistics]
+
+
+@attr.s
+class ImageData:
+    """Image Data class."""
+
+    data: numpy.ndarray = attr.ib()
+    mask: numpy.ndarray = attr.ib()
+    assets: Optional[List[str]] = attr.ib(default=None)
+
+    def __iter__(self):
+        """Allow for variable expansion (``arr, mask = ImageData``)"""
+        for i in (self.data, self.mask):
+            yield i
+
+    def as_masked(self):
+        """return a numpy masked array."""
+        data = numpy.ma.array(self.data)
+        data.mask = self.mask == 0
+        return data
+
+    def data_as_image(self) -> numpy.ndarray:
+        """return a numpy masked array."""
+        return reshape_as_image(self.data)
+
+    @property
+    def width(self) -> int:
+        """Width of the data array."""
+        return self.data.shape[2]
+
+    @property
+    def height(self) -> int:
+        """Height of the data array."""
+        return self.data.shape[1]
+
+    @property
+    def count(self) -> int:
+        """Number of band."""
+        return self.data.shape[0]
+
+    def post_process(
+        self,
+        in_range: Optional[Tuple[NumType, NumType]] = None,
+        out_range: Optional[Tuple[NumType, NumType]] = None,
+        color_formula: Optional[str] = None,
+    ):
+        """Post-process image data."""
+        if in_range:
+            rescale_arr = tuple(_chunks(in_range, 2))
+            if len(rescale_arr) != self.count:
+                rescale_arr = ((rescale_arr[0]),) * self.count
+
+            for bdx in range(self.count):
+                self.data[bdx] = numpy.where(
+                    self.mask,
+                    linear_rescale(
+                        self.data[bdx], in_range=rescale_arr[bdx], out_range=out_range
+                    ),
+                    0,
+                )
+            self.data = self.data.astype(numpy.uint8)
+
+        if color_formula:
+            self.data[self.data < 0] = 0
+            for ops in parse_operations(color_formula):
+                self.data = scale_dtype(ops(to_math_type(self.data)), numpy.uint8)
```

Comment:
This adds rio-color dependency

Reply:
Is this the only thing we're using from `rio-color`? I'm wary of adding so many new dependencies unless really necessary. For example some people put in work to make a conda package for rio-tiler 1.x. It'll be much harder to make a Conda package for rio-tiler 2.x ...; `rio-color` isn't on Conda

Reply:
I though adding rio-tiler (color_formula) could be nice (to remove downstream code in titiler or other dynamic tiler using rio-tiler).

about the conda problem ... well 🤷‍♂️